pkg/collector/corechecks/servicediscovery/apm/detect.go

```diff
@@ -134,48 +138,49 @@ func pythonDetector(args []string, envs map[string]string) Instrumentation {
 	return None
 }
 
+// isNodeInstrumented parses the provided `os.File` trying to find an
+// entry for APM NodeJS instrumentation. Returns true if finding such
+// an entry, false otherwise.
+func isNodeInstrumented(f *os.File) bool {
+	const readLimit = 16 * 1024 // Read 16KiB max
+
+	limitReader := io.LimitReader(f, readLimit)
+	bufferedReader := bufio.NewReader(limitReader)
+
+	return nodeAPMCheckRegex.MatchReader(bufferedReader)
```

Comment:
Does this match correctly if the string is split between two reads of the buffer?

Reply:
It should be working. I tried replacing the `NewReader` call with a `NewReaderSize` with a size of 1 byte, and it worked fine.